javaparser-core/src/main/java/com/github/javaparser/resolution/declarations/ResolvedTypeParameterDeclaration.java

```diff
@@ -128,15 +128,15 @@ default boolean declaredOnConstructor() {
      * This is unsupported because there is no package for a Type Parameter, only for its container.
      */
     default String getPackageName() {
-        throw new UnsupportedOperationException();
+        throw new UnsupportedOperationException(String.format("Unable to get package name for %s", this.getClass().getCanonicalName()));
     }
 
     /**
      * The class(es) wrapping the type bound(s).
      * This is unsupported because there is no class for a Type Parameter, only for its container.
      */
     default String getClassName() {
-        throw new UnsupportedOperationException();
+        throw new UnsupportedOperationException(String.format("Unable to get class name for %s", this.getClass().getCanonicalName()));
```

Comment:
Good! 

It would be helpful to also explain **WHY** it is unsupported -- some of this information is included in the javadoc above :)